examples/appprotect-dos/README.md

```diff
@@ -0,0 +1,67 @@
+# NGINX App Protect Dos Support
+
+In this example we deploy the NGINX Plus Ingress controller with [NGINX App Protect Dos](https://www.nginx.com/products/nginx-app-protect-dos/), a simple web application and then configure load balancing and DOS protection for that application using the Ingress resource.
+
+## Running the Example
+
+## 1. Deploy the Ingress Controller
+
+1. Follow the installation [instructions](https://docs.nginx.com/nginx-ingress-controller/installation) to deploy the Ingress controller with NGINX App Protect Dos.
+
+2. Save the public IP address of the Ingress controller into a shell variable:
+    ```
+    $ IC_IP=XXX.YYY.ZZZ.III
+    ```
+3. Save the HTTPS port of the Ingress controller into a shell variable:
+    ```
+    $ IC_HTTPS_PORT=<port number>
+    ```
+
+## 2. Deploy the Webapp Application
+
+Create the webapp deployment and service:
+```
+$ kubectl create -f webapp.yaml
+```
+
+## 3. Configure Load Balancing
+1. Create the syslog service and pod for the App Protect Dos security logs:
+    ```
+    $ kubectl create -f syslog.yaml
+    ```
+2. Create a secret with an SSL certificate and a key:
+    ```
+    $ kubectl create -f webapp-secret.yaml
+    ```
+3. Create the App Protect Dos Protected Resource:
+    ```
+    $ kubectl create -f apdos-protected.yaml
+    ```
+4. Create the App Protect Dos policy and log configuration:
+    ```
+    $ kubectl create -f apdos-policy.yaml
+    $ kubectl create -f apdos-logconf.yaml
+    ```
+5. Create an Ingress Resource:
+
+    ```
+    $ kubectl create -f webapp-ingress.yaml
+    ```
+    Note the App Protect Dos annotation in the Ingress resource. This enables DOS protection by specifying the DOS protected resource configuration that applies to this Ingress.
+
+## 4. Test the Application
+
+1. To access the application, curl the Webapp service. We'll use `curl`'s --insecure option to turn off certificate verification of our self-signed
+certificate and the --resolve option to set the Host header of a request with `webapp.example.com`
+
+    Send a request to the application::
+    ```
+    $ curl --resolve webapp.example.com:$IC_HTTPS_PORT:$IC_IP https://webapp.example.com:$IC_HTTPS_PORT/ --insecure
```

Comment:
Following this example didn't work for me - checking the logs I see an error:

```
I1213 12:13:51.045794       1 event.go:285] Event(v1.ObjectReference{Kind:"Ingress", Namespace:"default", Name:"webapp-ingress", UID:"26150234-6b51-4c40-b2a6-a45495e081f4", APIVersion:"networking.k8s.io/v1", ResourceVersion:"10457", FieldPath:""}): type: 'Warning' reason: 'AddedOrUpdatedWithError' Configuration for default/webapp-ingress was added or updated ; but was not applied: Error adding or updating ingress default/webapp-ingress: Error generating Ingress Config default-webapp-ingress: template: nginx-plus.ingress.tmpl:89:15: executing "nginx-plus.ingress.tmpl" at <$server.AppProtectDosMonitorUri>: can't evaluate field AppProtectDosMonitorUri in type version1.Server
```

Reply:
fixed